src/app/petfinder/petfinder-service/petfinder.service.ts

```diff
@@ -19,29 +19,28 @@ export class PetfinderService implements OnDestroy {
   };
 
   private refreshTokenSubject = new BehaviorSubject<void>(undefined);
-  private refreshToken$ = this.refreshTokenSubject.asObservable();
 
   private getToken$ = this.http.post<Token>(this.baseUrl + '/v2/oauth2/token',
-    `grant_type=client_credentials&client_id=${this.apiKey}&client_secret=${this.apiSecret}`, this.tokenHeader);
-
-  token$ = combineLatest([
-    this.getToken$,
-    this.refreshToken$
-  ]).pipe(
-    mergeMap(() => this.getToken$)
+    `grant_type=client_credentials&client_id=${this.apiKey}&client_secret=${this.apiSecret}`, this.tokenHeader).pipe(
```

Comment:
I think this is mock auth but normally we would use a different grant type that would not rquire us to send secrets in javascript.